lib/tf_lock/tf_lock_acquire.py

```diff
@@ -62,6 +70,13 @@ def tf_lock_acquire(root_module: Path) -> ExitCode:
         # start over
 
 
+def is_pr_closed(gh_url: str) -> bool:
+    status = sh.stdout(
+        ("gh", "pr", "view", gh_url, "--json", "state", "--jq", ".state")
+    )
+    return status in ["CLOSED", "MERGED"]
```

Comment:
Hmm. This seems familiar. It looks like we have three very similar things in the codebase already:

```
lib/tacos/conflict_detection.py:    if state in ("MERGED", "CLOSED"):
lib/tf_lock/release.py:        return state in ("MERGED", "CLOSED")
spec/lib/gh/pr.py:        return status in ["CLOSED", "MERGED"]
```

Please unify, as much as you can.

---
Suggestion: break out a function of `pr_url` in `spec/lib/gh/pr.py`, then you can import and reuse it here.


```console
$ git apply <<EOF
diff --git a/spec/lib/gh/pr.py b/spec/lib/gh/pr.py
index 2a0594d..8efd4dc 100644
--- a/spec/lib/gh/pr.py
+++ b/spec/lib/gh/pr.py
@@ -105,10 +105,7 @@ class PR:
         return since

     def is_closed(self) -> bool:
-        status = sh.stdout(
-            ("gh", "pr", "view", self.url, "--json", "state", "--jq", ".state")
-        )
-        return status in ["CLOSED", "MERGED"]
+        return pr_is_closed(self.url)

     def approve(
         self, app_installation: app.Installation, now: datetime | None = None
@@ -143,7 +140,7 @@ class PR:
     def merge(self) -> datetime:
         sh.banner("merging PR")
         since = mknow()
-        sh.run(("gh", "pr", "merge", self.url, "--squash"))
+        sh.run(("gh", "pr", "merge", self.url, "--auto", "--squash"))
         return since

     def labels(self) -> Sequence[Label]:
@@ -243,6 +240,13 @@ class PR:
                 yield run


+def pr_is_closed(pr_url: URL) -> bool:
+    status = sh.stdout(
+        ("gh", "pr", "view", pr_url, "--json", "state", "--jq", ".state")
+    )
+    return status in ["CLOSED", "MERGED"]
+
+
 def commit_and_push(branch: Branch, message: object = None) -> datetime:
     since = mknow()
     sh.run(("git", "commit", "-qam", message))
EOF
```

Reply:
pr.py is probably the place to go to reuse